### PR TITLE
Remove legacy grouping props

### PR DESCRIPTION
**Pull request by amanmahajan7**
Cell selection on the grouped row is off on the grouped row but that is how [legacy ](https://w741nqnxv7.csb.app/)worked so I am not changing it. Let's revisit when we add the new grouping API

### CODE REVIEW

File: src/RowRenderer.tsx [hunk 1]
```diff
@@ -1,40 +1,26 @@
...
 
-interface IRowRendererProps<R, SR> extends SharedDataGridProps<R, SR> {
-  viewportColumns: readonly CalculatedColumn<R, SR>[];
-  lastFrozenColumnIndex: number;
-  rowIdx: number;
-  row: R;
-  eventBus: EventBus;
-  isRowSelected: boolean;
-}
+type IRowRendererProps<R, SR> = RowRendererProps<R, SR> & SharedDataGridProps<R, SR>;
```
Comment (qili26) [inline src/RowRenderer.tsx:12]: Do we still need this? Or we should use it at ln#23

File: src/RowRenderer.tsx [hunk 2]
```diff
@@ -1,40 +1,26 @@
...
   row,
-  rowGroupRenderer,
   rowRenderer,
   isRowSelected,
-  onRowClick,
-  onRowExpandToggle
-}: IRowRendererProps<R, SR>) {
-  const { __metaData } = row as RowData;
+  onRowClick
+}: RowRendererProps<R, SR> & SharedDataGridProps<R, SR>) {
```
Comment (amanmahajan7) [inline src/RowRenderer.tsx:23]: I wonder if we should use memo on the `Row` component and delete `RowRenderer`. Thoughts?

Comment (qili26) [reply src/RowRenderer.tsx:23]: We still want to have the ability to customize the rowrenderer, right? I believe that's the reason why we kept this layer.

File: stories/demos/LegacyGrouping.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+  }
+
+  return gridRows;
+}
+
+export default function LegacyGrouping() {
+  const [rows] = useState(createRows);
+  const [groupByColumns] = useState(['priority', 'issueType']);
+  const [expandedGroups, setExpandedGroups] = useState(new Set(['Low', 'Low_Epic']));
```
Comment (qili26) [inline stories/demos/LegacyGrouping.tsx:87]: Don't we need this `() => new Set(['Low', 'Low_Epic'])` in `useState`?

File: stories/demos/components/RowRenderers/GroupRowRenderer.less [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    color: #777;
+  }
+}
+
+.rdg-row-default-group {
+  width: var(--row-width);
+  height: var(--row-height);
+  line-height: var(--row-height);
+  padding: 0 8px;
+  border-bottom: 1px solid #ddd;
```
Comment (qili26) [inline stories/demos/components/RowRenderers/GroupRowRenderer.less:14]: `border-bottom` seems like a common variable, I'm wondering if we want to create a css variable for this.

File: package.json [hunk 5]
```diff
@@ -74,6 +75,7 @@
     "jest-environment-jsdom-sixteen": "^1.0.3",
     "less": "^3.11.1",
     "less-loader": "^5.0.0",
+    "lodash.groupby": "^4.6.0",
```
Comment (nstepien) [inline package.json:78]: Use the `lodash` package directly instead. It's more likely to already be installed by other deps, and we get less duplicate code (`lodash/groupBy` + `lodash.groupby`).
If you're worried about bundle size, then `import ... from 'lodash/...'` should be used instead.

File: src/RowRenderer.tsx (hunk 2 above)
Comment (nstepien) [inline src/RowRenderer.tsx:23]:
```suggestion
}: IRowRendererProps<R, SR>) {
```

Comment (nstepien) [reply src/RowRenderer.tsx:23]: Let's just delete this module, memoize `Row`, and use the Row/rowRenderer directly.
Custom row renderers can memo themselves.

File: stories/demos/LegacyGrouping.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  const [columnKey, ...remainingColumnKeys] = columnKeys;
+  const groupedRows = groupBy(rows, columnKey);
+  const groupedKeys = Object.keys(groupedRows);
+  for (const key of groupedKeys) {
+    const groupKey = parentKey ? `${parentKey}_${key}` : key;
+    const isExpanded = expandedGroups.has(groupKey);
+    const rowGroupHeader = {
+      groupKey,
+      name: key,
+      __metaData: {
```
Comment (nstepien) [inline stories/demos/LegacyGrouping.tsx:67]: I think it'd be better to refactor this to a more realistic data structure, maybe something like
```ts
interface RowGroup {
  type: 'group';
  // ...
}

interface RowData {
  type: 'data';
  // ...
}

type Row = RowGroup | RowData;
```

Not worth our time right now though.

File: stories/demos/LegacyGrouping.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+      gridRows.push(...groupByColumn(groupedRows[key], remainingColumnKeys, expandedGroups, treeDepth + 1, key));
+    }
+  }
+
+  return gridRows;
+}
+
+export default function LegacyGrouping() {
+  const [rows] = useState(createRows);
+  const [groupByColumns] = useState(['priority', 'issueType']);
```
Comment (nstepien) [inline stories/demos/LegacyGrouping.tsx:86]: Let's move this outside the component.
```suggestion
  const groupByColumns = ['priority', 'issueType'];
```

File: stories/demos/LegacyGrouping.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  const [expandedGroups, setExpandedGroups] = useState(new Set(['Low', 'Low_Epic']));
+
+  const gridRows = useMemo(() => {
+    return groupByColumn(rows, groupByColumns, expandedGroups);
+  }, [rows, groupByColumns, expandedGroups]);
```
Comment (nstepien) [inline stories/demos/LegacyGrouping.tsx:91]:
```suggestion
  }, [rows, expandedGroups]);
```

File: stories/demos/LegacyGrouping.tsx (hunk 6 above)
Comment (amanmahajan7) [reply stories/demos/LegacyGrouping.tsx:85]: Simplified types a bit

File: stories/demos/LegacyGrouping.tsx (hunk 3 above)
Comment (amanmahajan7) [reply stories/demos/LegacyGrouping.tsx:131]: It is not required but I have made the change

**Pull request merged**